plugin/src/main/java/com/microsoft/java/bs/gradle/plugin/SourceSetsModelBuilder.java

```diff
@@ -127,89 +140,87 @@ public Object buildAll(String modelName, Project rootProject) {
         continue;
       }
 
-      // dependencies
       sourceSets.forEach(sourceSet -> {
         DefaultGradleSourceSet gradleSourceSet = sourceSetMap.get(sourceSet);
         if (gradleSourceSet == null) {
           return;
         }
+        // dependencies
         DependencyCollector collector = new DependencyCollector(project,
             exclusionFromDependencies);
         collector.collectByConfigurationNames(getClasspathConfigurationNames(sourceSet));
         gradleSourceSet.setModuleDependencies(collector.getModuleDependencies());
         gradleSourceSet.setProjectDependencies(collector.getProjectDependencies());
+
+        // scala.  Requires module dependencies to resolve version info.
+        ScalaCompile scalaCompile = getScalaCompileTask(project, sourceSet);
+        if (scalaCompile != null) {
+          GradleModuleDependency scalaLibraryDependency =
+              getScalaLibraryDependency(gradleSourceSet.getModuleDependencies());
+          if (scalaLibraryDependency != null) {
+            gradleSourceSet.setIsScala(true);
+            gradleSourceSet.setScalaOrganization(getScalaOrganization(scalaLibraryDependency));
+            gradleSourceSet.setScalaVersion(getScalaVersion(scalaLibraryDependency));
```

Comment:
Hi @Arthurm1, just a question, no action needs to take.

Today when I was reading the spec of BSP, I found following statements:

```
a single configuration in a single project that contains both Java and Scala sources maps to one BuildTarget.
a project with crossScalaVersions 2.11 and 2.12 containing main and test configuration in each will have 4 build targets.
a Scala 2.11 and 2.12 cross-built project for Scala.js and the JVM with main and test configurations will have 8 build targets.
```

I'm pretty sure that current implementation can apply the first one. What about the second and third?

Reply:
This is really for other build tools that support BSP e.g. SBT, Mill, Bazel.  It's not possible to do cross versioning in Gradle with the built in Scala plugin.  Unless there's a 3rd party plugin somewhere around that I'm not aware of.

Reply:
Ah, I see. Thank you!